src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -2256,6 +2259,291 @@ def isomorphism_to(self, other, *, conjugator=False):
+        \nmid 2i^2j^2`.
+
+        Used as a subroutine in :meth:`_P1_to_ideals`.
+
+        EXAMPLES::
+
+            sage: ell = 3
+            sage: B.<i, j, k> = QuaternionAlgebra(-5, -11)
+            sage: M = Matrix([[150, 0, 0, 0], [0, 106, 0, 32], [0, 53, 75, 16], [75, -14, 0, 17]])
+            sage: I = M * vector(B.basis()) / 150; I
+            doctest:warning ... UserWarning: ...
+            (1, 53/75*i + 16/75*k, 53/150*i + 1/2*j + 8/75*k, 1/2 - 7/75*i + 17/150*k)
+            sage: O = B.quaternion_order(I)
+            sage: O.basis_matrix().denominator() % ell == 0
+            True
+
+            sage: O1, a = O._denominator_coprime(ell)
+            sage: a  # random
+            -2/5 - 37/75*i + 66/5*j - 1499/75*k
+            sage: list(O1.basis()) == [a * g * ~a for g in O.basis()]
+            True
+            sage: O1.basis_matrix().denominator() % ell == 0
+            False
+        """
+        # assert gcd(ell, 2 * self.gen(0)**2 * self.gen(1)**2) != 0
+        B = self.quaternion_algebra()
+
+        if self.basis_matrix().denominator() % ell != 0:
+            return self, B.one()
+
+        O0 = B.maximal_order()  # <- The order with denominator 2q
+        I_conn = O0 * self
+        I_conn *= 1 / I_conn.norm()
+
+        while True:
+            alpha = sum([ZZ.random_element(-100, 100)*beta for beta in I_conn.basis()])
+            if gcd(alpha.reduced_norm(), I_conn.norm()**2) == I_conn.norm():
+                break
+
+        alpha /= I_conn.norm()
+        O_new = B.quaternion_order([alpha * g * ~alpha for g in self.basis()])
+        return O_new, alpha
+
+    def _P1_to_ideals(self, ell):
+        r"""
+        Return a map from the projective line over `\ZZ / \ell` to the
+        left ideals of norm `\ell`. This method is useful for enumerating or
+        generating ideals of a given norm.
+
+        This method implements the bijection given by lemma 7.2 in [KV2010]_.
+
+        EXAMPLES::
+
+            sage: B = QuaternionAlgebra(next_prime(2**50))
+            sage: O = B.maximal_order()
+            sage: ell = 13
+            sage: mp = O._P1_to_ideals(ell)
+            sage: mp(0, 1)
+            Fractional ideal (1/2 + 1/2*j + 10*k, 1/2*i + 3*j + 1/2*k, 13*j, 13*k)
+            sage: mp(1, 3)
+            Fractional ideal (1/2 + 5/2*j + 9*k, 1/2*i + 4*j + 5/2*k, 13*j, 13*k)
+
+        This method is used internally to enumerate :meth:`all_left_ideals_of_norm`::
+
+            sage: image = [mp(0, 1)] + [mp(1, k) for k in range(ell)]
+            sage: set(O.all_left_ideals_of_norm(ell)) == set(image)
+            True
+        """
+        B = self.quaternion_algebra()
+        if ell in B.ramified_primes():
+            raise ValueError(f"ell(={ell}) is not split")
+
+        if not self.quadratic_form().is_positive_definite():
+            raise TypeError("the quaternion algebra must be positive definite")
+
+        phi = B.modp_splitting_map(ell)
+        F = GF(ell)
```

Comment:
Is there ever the case where `ell` might be very large? Is it worth adding a `names` to this call to stop using the Conway polynomial?

```suggestion
        F = GF(ell, names="a")
```